presto-hive/src/main/java/com/facebook/presto/hive/CopyOnWriteConfiguration.java

```diff
@@ -0,0 +1,524 @@
+            throws IOException
+    {
+        return config.getLocalPath(dirsProp, path);
+    }
+
+    @Override
+    public File getFile(String dirsProp, String path)
+            throws IOException
+    {
+        return config.getFile(dirsProp, path);
+    }
+
+    @Override
+    public URL getResource(String name)
+    {
+        return config.getResource(name);
+    }
+
+    @Override
+    public InputStream getConfResourceAsInputStream(String name)
+    {
+        return config.getConfResourceAsInputStream(name);
+    }
+
+    @Override
+    public Reader getConfResourceAsReader(String name)
+    {
+        return config.getConfResourceAsReader(name);
+    }
+
+    @Override
+    public Set<String> getFinalParameters()
+    {
+        return config.getFinalParameters();
+    }
+
+    @Override
+    public int size()
+    {
+        return config.size();
+    }
+
+    @Override
+    public void clear()
+    {
+        config.clear();
+    }
+
+    @Override
+    public Iterator<Map.Entry<String, String>> iterator()
+    {
+        return config.iterator();
+    }
+
+    @Override
+    public ClassLoader getClassLoader()
+    {
+        return config.getClassLoader();
+    }
+
+    @Override
+    public void readFields(DataInput in)
+            throws IOException
+    {
+        config.readFields(in);
+    }
+
+    @Override
+    public void write(DataOutput out)
+            throws IOException
+    {
+        config.write(out);
+    }
+
+    @Override
+    public void writeXml(OutputStream out)
+            throws IOException
+    {
+        config.writeXml(out);
+    }
+
+    @Override
+    public void writeXml(Writer out)
+            throws IOException
+    {
+        config.writeXml(out);
+    }
+
+    @Override
+    public Map<String, String> getValByRegex(String regex)
+    {
+        return config.getValByRegex(regex);
+    }
+
+    @Override
+    public String toString()
+    {
+        return config.toString();
+    }
+
+    @Override
+    public void addResource(String name)
+    {
+        config.addResource(name);
+    }
+
+    @Override
+    public void addResource(URL url)
+    {
+        config.addResource(url);
+    }
+
+    @Override
+    public void addResource(Path file)
+    {
+        config.addResource(file);
+    }
+
+    @Override
+    public void addResource(InputStream in)
+    {
+        config.addResource(in);
+    }
+
+    @Override
+    public void addResource(InputStream in, String name)
+    {
+        config.addResource(in, name);
+    }
+
+    @Override
+    public void addResource(Configuration conf)
+    {
+        config.addResource(conf);
+    }
+
+    @Override
+    public synchronized void reloadConfiguration()
+    {
+        config.reloadConfiguration();
+    }
+
+    @Override
+    public synchronized String[] getPropertySources(String name)
+    {
+        return config.getPropertySources(name);
+    }
+
+    private void checkAndSet(Runnable action)
+    {
+        if (!isMutable) {
```

Comment:
Handle the isMutable case first in a IF branch. To reduce the indent for more lines of code to improve readability.